File: 19.12.25/Two_sum.java
import java.util.ArrayList;
import java.util.Arrays;
class Solution {

public static void main(String[]args){
    Scanner sc= new Scanner(System.in);
    System.out.println("enter the size of array:::::::::");
    int size=sc.nextInt();
    System.out.println("enter the traget:::::::::");
    int target=sc.nextInt();
    int[] arr= new int[size];
    System.out.println("enter the array:::::::::");
    for(int i=0; i<size;i++){
        arr[i]=sc.nextInt();
    }
    System.out.println("final  array is :: "+arr);
    Solution s = new Solution();
    s.twoSum(arr,target);
  }
   public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> map= new HashMap<>();

    for(int i=0; i<nums.length;i++){
        int complement=target-nums[i];
        if(map.containsKey(complement)){
            return new int[]{map.get(complement),i};
        }
        map.put(nums[i], i);
    }

    return new int[0];
   }

}

File: 01.01.26/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        i = 0
        b = 1
        while True:
            ans = nums[i] + nums[b]
            if ans == target:
                return [i, b]
            elif b != len(nums)-1:
                b += 1
            else:
                i += 1
                b = i + 1
            
